text/0911-const-fn.md

```diff
@@ -172,6 +172,18 @@ const fn arithmetic_ops<T: Int>() -> [fn(T, T) -> T; 4] {
 }
 ```
 
+`const` functions can also be unsafe, allowing construction of types that require
+invariants to be maintained (e.g. `std::ptr::Unique` requires a non-null pointer)
+```rust
+struct OptionalInt(u32);
+impl OptionalInt {
+    /// Value must be non-zero
+    unsafe const fn new(val: u32) -> OptionalInt {
```

Comment:
Hmm... both could be allowed, but would complicate the parser (if it is to correctly disallow `unsafe const unsafe fn`). I'll yeild to the language team on the final decision, but I'm leaning towards keeping a fixed order.